internal/sequencerapi/api.go

```diff
@@ -83,6 +86,14 @@ func (s *sendRawTxCond) SendRawTransactionConditional(ctx context.Context, txByt
 		}
 	}
 
+	// enforce rate limit on the cost to be observed
+	if err := s.costLimiter.WaitN(ctx, cost); err != nil {
```

Comment:
Just to be sure, you intend to wait *after* doing all the expensive checks above? Is that intended?

Reply:
Yes perfect.